Using `assert` on mutable receiver var that implements an interface and returns an Option results in CGen error

### Describe the bug

Running below V code with `v run .` results in:

```
=================
/dev/shm/v_1000/ifacebug.8486332335434900259.tmp.c:12883: error: cannot convert 'struct main__Item' to 'unsigned long'
...
==================
(Use `v -cg` to print the entire error message)
```

### Reproduction Steps

V Run `bug.v`:
```v
@[heap]
struct UI {}

pub fn (ui &UI) draw_rect() {
	println('[]')
}

@[heap]
pub interface Node {
	id u64
	draw()
mut:
	ui &UI
	init(ui &UI) !
}

@[heap]
pub struct Item {
pub:
	id u64 = 1
mut:
	ui   &UI = unsafe { nil }
	body []&Node
}

pub fn (mut i Item) init(ui &UI) ! {
	assert i != unsafe { nil } // This assert generates a C gen error
	i.ui = ui
	for mut child in i.body {
		child.init(ui)!
	}
}

pub fn (i &Item) draw() {
	assert i != unsafe { nil }
	for child in i.body {
		child.draw()
	}
}

@[heap]
pub struct Rectangle {
	Item
pub mut:
	field f32
}

pub fn (r &Rectangle) draw() {
	assert r != unsafe { nil }
	r.ui.draw_rect()
	r.Item.draw()
}

fn main() {
	ui := &UI{}
	mut rect := &Rectangle{}

	rect.init(ui)!

	rect.draw()
}
```

### Expected Behavior

Program compiles cleanly and outputs `[]`

### Current Behavior

```
=================
/dev/shm/v_1000/ifacebug.8486332335434900259.tmp.c:12883: error: cannot convert 'struct main__Item' to 'unsigned long'
...
==================
(Use `v -cg` to print the entire error message)
```

### Possible Solution

Removing the assert statement makes the code compile and run

### Additional Information/Context

_No response_

### V version

V 0.4.3 d474f69

### Environment details (OS name and version, etc.)

Arch linux based / EndeavourOS
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -1053,7 +1053,7 @@ fn (mut g Gen) gen_plain_infix_expr(node ast.InfixExpr) {
 		typ_str := g.typ(node.promoted_type)
 		g.write('(${typ_str})(')
 	}
-	if node.left_type.is_ptr() && node.left.is_auto_deref_var() {
+	if node.left_type.is_ptr() && node.left.is_auto_deref_var() && !node.right_type.is_pointer() {
 		g.write('*')
 	} else if !g.inside_interface_deref && node.left is ast.Ident
 		&& g.table.is_interface_var(node.left.obj) {
@@ -1065,7 +1065,7 @@ fn (mut g Gen) gen_plain_infix_expr(node ast.InfixExpr) {
 	}
 	g.expr(node.left)
 	g.write(' ${node.op.str()} ')
-	if node.right_type.is_ptr() && node.right.is_auto_deref_var() {
+	if node.right_type.is_ptr() && node.right.is_auto_deref_var() && !node.left_type.is_pointer() {
 		g.write('*')
 		g.expr(node.right)
 	} else {

--- a/vlib/v/tests/infix_expr_in_mut_receiver_method_test.v
+++ b/vlib/v/tests/infix_expr_in_mut_receiver_method_test.v
@@ -1,0 +1,62 @@
+@[heap]
+struct UI {}
+
+pub fn (ui &UI) draw_rect() {
+	println('[]')
+}
+
+@[heap]
+pub interface Node {
+	id u64
+	draw()
+mut:
+	ui &UI
+	init(ui &UI) !
+}
+
+@[heap]
+pub struct Item {
+pub:
+	id u64 = 1
+mut:
+	ui   &UI = unsafe { nil }
+	body []&Node
+}
+
+pub fn (mut i Item) init(ui &UI) ! {
+	assert i != unsafe { nil } // This assert generates a C gen error
+	i.ui = ui
+	for mut child in i.body {
+		child.init(ui)!
+	}
+}
+
+pub fn (i &Item) draw() {
+	assert i != unsafe { nil }
+	for child in i.body {
+		child.draw()
+	}
+}
+
+@[heap]
+pub struct Rectangle {
+	Item
+pub mut:
+	field f32
+}
+
+pub fn (r &Rectangle) draw() {
+	assert r != unsafe { nil }
+	r.ui.draw_rect()
+	r.Item.draw()
+}
+
+fn test_infix_expr_in_mut_receiver_method() {
+	ui := &UI{}
+	mut rect := &Rectangle{}
+
+	rect.init(ui)!
+
+	rect.draw()
+	assert true
+}

